Login who owns database should always be dbo also without sysadmin role

wboettge opened the pull request:
### Description

In SQL Server, the login that creates a database is automatically the
database owner. This means that the login can always access the
databases they create.

However, previously in babelfish, if a login is stripped of the sysadmin
role after creating a database, and the login does not have a
user associated with the database, then the login can no longer access
the database. Now, when a login attempts to access and there
is not an associated user, the login will be considered the dbo and
allowed access.

Task: BABEL-3548

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

wboettge committed:
Login owning db should be dbo even w/out sysadmin

In SQL Server, the login that creates a database is automatically the
database owner. This means that the login can always access the
databases they create.

However, previously in babelfish, if a login is stripped of the sysadmin
role after creating a database, and the login does not have a
user associated with the database, then the login can no longer access
the database. Now, when a login attempts to access and there
is not an associated user, the login will be considered the dbo and
allowed access.

Task: BABEL-3548

Signed-off-by: Walt Boettge <wboettge@amazon.com>

wboettge committed:
Also allow logins to drop owned dbs

In addition to being able to access owned dbs without the sysadmin role,
logins can now drop databases they previously created.

Signed-off-by: Walt Boettge <wboettge@amazon.com>

xhfanhe commented on contrib/babelfishpg_tsql/src/dbcmds.c:
```diff
@@ -832,3 +834,22 @@ drop_related_bbf_namespace_entries(int16 dbid)
+	HeapTuple 			tuple;
+	Form_sysdatabases 	sysdb;
+
+	  tuple = SearchSysCache1(SYSDATABASENAME, CStringGetTextDatum(dbname));
```
Please fix the indentation here

xhfanhe commented on test/JDBC/input/BABEL-3548.mix:
```diff
@@ -0,0 +1,62 @@
+-- tsql
+create login test_login1 with password='123';
```
For all the created objects in this file, please add a `babel_3548` at the beginning of the object name, in this way if something went wrong, we can easily locate where the error happens

wboettge committed:
Rename test objects and fix indents

Signed-off-by: Walt Boettge <wboettge@amazon.com>

xhfanhe commented on contrib/babelfishpg_tsql/src/dbcmds.c:
```diff
@@ -832,3 +834,22 @@ drop_related_bbf_namespace_entries(int16 dbid)
+	tuple = SearchSysCache1(SYSDATABASENAME, CStringGetTextDatum(dbname));
+
+	if (!HeapTupleIsValid(tuple))
+		return InvalidDbid;
```
I feel it doesn't make much sense to return InvalidDbid here. We can consider to throw an error if the given db name is invalid. And please add some comment ahead of this function saying that caller is responsible for validating db name before calling this function.

wboettge committed:
Fix missing dbname error and switch to strncmp

If searching for the dbname gives an invalid tuple, get_owner_of_db now
throws an UNDEFINED_DATABASE error instead of returning InvalidDbId.

When checking if the login matches the dbowner, strncmp is used instead
of strcmp, with NAMEDATALEN as the max number of characters to compare.

Signed-off-by: Walt Boettge <wboettge@amazon.com>

xhfanhe commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -936,22 +937,25 @@ get_user_for_database(const char *db_name)
 
 	login = GetUserNameFromId(GetSessionUserId(), false);
 	user = get_authid_user_ext_physical_name(db_name, login);
+	login_is_db_owner = 0 == strncmp(login, get_owner_of_db(db_name), NAMEDATALEN);
```
I tend to think it's better to use strcmp here, is there any specific considerations here?
